node_modules/oae-principals/lib/rest.group.js

```diff
@@ -377,3 +377,156 @@ OAE.tenantRouter.on('post', '/api/group/:groupId/picture', function(req, res) {
+        }
+        return res.status(200).end();
+    });
+});
+
+/**
+ * @REST postGroupGroupIdRequestJoinRemove
+ *
+ * Remove a request from a user
+ *
+ * @Server      tenant
+ * @Method      POST
+ * @Path        /group/{groupId}/request-join/remove
+ * @PathParam   {string}        groupId                 The id of the group to join
+ * @Return      {void}
+ * @HttpResponse                200                     Request removed
+ * @HttpResponse                401                     You have to be logged in to be able to leave a group
+ * @HttpResponse                402                     Invalid groupId specified
+ */
+OAE.tenantRouter.on('delete', '/api/group/:groupId/request-join/remove', function(req, res) {
+
+    PrincipalsAPI.removeJoinGroupByRequest(req.ctx, req.params.groupId, function(err) {
+        if (err) {
+            return res.status(err.code).send(err.msg);
+        }
+        return res.status(200).end();
+    });
+});
+
+/**
+ * @REST postGroupGroupIdRequestJoinAccept
+ *
+ * Accept a request
+ *
+ * @Server      tenant
+ * @Method      POST
+ * @Path        /group/{groupId}/request-join/accept
+ * @PathParam   {string}        groupId                 The id of the group to join
+ * @PathParam   {string}        principalId             The id of the principal who wants to join this group
+ * @PathParam   {string}        role                    The role ask by the user who wants to join the group
+ * @Return      {void}
+ * @HttpResponse                200                     Request accepted
+ * @HttpResponse                400                     Invalid role specified
+ * @HttpResponse                401                     You have to be logged in to be able to leave a group
+ * @HttpResponse                402                     Invalid groupId specified
+ */
+OAE.tenantRouter.on('post', '/api/group/:groupId/request-join/accept', function(req, res) {
+
+    PrincipalsAPI.acceptJoinGroupByRequest(req.ctx, req.params.groupId, req.body.principalId, req.body.role, function(err) {
+        if (err) {
+            return res.status(err.code).send(err.msg);
+        }
+        return res.status(200).end();
+    });
+});
+
+/**
+ * @REST postGroupGroupIdRequestJoinReject
+ *
+ * Reject a request
+ *
+ * @Server      tenant
+ * @Method      POST
+ * @Path        /group/{groupId}/request-join/reject
+ * @PathParam   {string}        groupId                 The id of the group to join
+ * @PathParam   {string}        principalId             The id of the principal who wants to join this group
+ * @Return      {void}
+ * @HttpResponse                200                     Request rejected
+ * @HttpResponse                401                     You have to be logged in to be able to leave a group
+ * @HttpResponse                402                     Invalid groupId specified
+ */
+OAE.tenantRouter.on('delete', '/api/group/:groupId/request-join/reject', function(req, res) {
```

Comment:
a reject is not a `DELETE` but instead a `PUT` because it will update a record. Having a table with a `status` field, we'll mark a request resolved or rejected by `PUT`ting because that will reflect on an update in CQL